driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -120,31 +124,33 @@ public void run() {
 
         Executor initExecutor = manager.cluster.manager.configuration.getPoolingOptions().getInitializationExecutor();
 
-        ListenableFuture<List<Void>> allConnectionsFuture = Futures.allAsList(connectionFutures);
+        ListenableFuture<List<Void>> allConnectionsFuture = Futures.successfulAsList(connectionFutures);
 
         final SettableFuture<Void> initFuture = SettableFuture.create();
-        Futures.addCallback(allConnectionsFuture, new FutureCallback<List<Void>>() {
+        Futures.addCallback(allConnectionsFuture, new MoreFutures.SuccessCallback<List<Void>>() {
             @Override
             public void onSuccess(List<Void> l) {
+                // Some of the connections might have failed, keep only the successful ones
+                ListIterator<Connection> it = connections.listIterator();
```

Comment:
`connections` is not the result of the "compound" future, it's a list we built previously with all the connections, before initializing them.

The result of `successfulAsList` is what gets passed to the callback (`List<Void> l`). It contains the results for the futures that have succeeded, and `null` for those that have failed. But since our futures are `Future<Void>`, the successful results are also `null`, so we can't do anything with `l`.